### PR TITLE
'ordNub' is now stable and lazy

### PR DESCRIPTION
**Pull request by jystic**
New implementation of `ordNub` which produces elements in their original order and is also lazy.

These two things are important for some QuickCheck stuff I'm doing where I want to filter out duplicate shrinks.

It requires a dependency on containers which we already have transitively via `bifunctors`.

/cc @nhibberd @markhibberd @charleso

### CODE REVIEW

File: ambiata-p.cabal [hunk 1]
```diff
@@ -15,8 +15,9 @@ description:           A set of standard preludes for Ambiata projects. \
 library
   build-depends:
                           base                        >= 4.6        && < 5
-                        , bifunctors                  >= 4.2        && <= 5.3
-                        , deepseq                     >= 1.3        && <= 1.5
+                        , bifunctors                  >= 4.2        && < 5.4
+                        , containers                  >= 0.5        && < 0.6
+                        , deepseq                     >= 1.3        && < 1.5
```
Comment (jystic) [inline ambiata-p.cabal]: Should probably be a separate PR, but I noticed when I added containers that we had `<=` on some upper bounds so I've corrected it.

Note that the latest version of `bifunctors` is `5.3`, and the latest version of `deepseq` is `1.4.2.0`. So the bound we had on `deepseq` was incorrect (my bad actually).

File: ambiata-p.cabal [hunk 2]
```diff
@@ -15,8 +15,9 @@ description:           A set of standard preludes for Ambiata projects. \
 library
   build-depends:
                           base                        >= 4.6        && < 5
-                        , bifunctors                  >= 4.2        && <= 5.3
-                        , deepseq                     >= 1.3        && <= 1.5
+                        , bifunctors                  >= 4.2        && < 5.4
+                        , containers                  >= 0.5        && < 0.6
```
Comment (markhibberd) [inline ambiata-p.cabal]: Not sure what would of lower bounds been here implicitly? This is probably ok, I don't think I have seen any containers 0.4 around for a while, but I am assuming this probably works with 0.4 as well? Is it worth being a little more generous just because it is `p`?

Comment (jystic) [inline ambiata-p.cabal]: Yeah completely agree

**Comment by markhibberd** [comment]
Makes sense to me.

👍  on 📗

**Comment by markhibberd** [comment]
(and agree deepseq change is legit)

**Comment by amosr** [comment]
any chance of an `nubSorted :: Eq a => [a] -> [a]` which takes an already sorted list and nubs only consecutive elements?
I suppose that is really just `concat . group` though, isn't it?

**Comment by jystic** [comment]
>  I suppose that is really just `concat . group` though, isn't it?

Yep, with `take 1` that would do it: `concat . take 1 . group`
